qmule/src/org/qcmg/qmule/bam/QSamToFastq.java

```diff
@@ -103,45 +108,92 @@
             "value is null then all bases left after trimming will be written.", optional=true)
     public Integer READ2_MAX_BASES_TO_WRITE;
 
-    @Argument(doc="If true, include non-primary alignments in the output.  Support of non-primary alignments in SamToFastq " +
-     "is not comprehensive, so there may be exceptions if this is set to true and there are paired reads with non-primary alignments.")
-    public boolean INCLUDE_NON_PRIMARY_ALIGNMENTS=false;
+    
+    //new argument 
+    @Argument(doc="If true, read id will be appended /1 for first of pair and /2 for second of pair. If false, read id will be as same as BAM record id.")
+    public boolean MARK_MATE = false;
 
-    private final Log log = Log.getInstance(QSamToFastq.class);
+    @Argument(doc="If true, set 'N' to fastq record base if SAM record missing base sequence; and then set '!' to base quality. If false, read base will be same as BAM record base, often is '*'.")
+    public boolean BASE_NULL_TO_N = true;
+    
+    @Argument(doc="If true, output a pair of fastq record, set base sequence 'N' and base quality '!' to the missing mate record . If false, output one fastq record only if the input SAM record missing mate.")
```

Comment:
"pair of fastq records" rather than "pair of fastq record"
"If false, output one fastq record (only if the input SAM record is missing it's mate)." rather than "If false, output one fastq record only if the input SAM record missing mate."